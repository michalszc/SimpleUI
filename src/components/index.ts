export * from "./Button";
export * from "./ColorPicker";
export * from './Switch';
export * from './FormLabel';