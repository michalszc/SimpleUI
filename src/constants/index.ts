export { 
    default as Colors,
    red,
    pink,
    purple,
    deepPurple,
    indigo,
    blue,
    lightBlue,
    cyan,
    teal,
    green,
    lightGreen,
    lime,
    yellow,
    amber,
    orange,
    deepOrange,
    brown,
    grey,
    blueGrey,
    black,
    white,
    transparent
} from './Colors';
export type { Color, IColors } from './Colors';
export { default as Palette } from './Palette';
export type { IPalette, PaletteColor } from './Palette';
export { default as Typography } from './Typography';
export type { ITypography, TypographyVariant } from './Typography';
export { default as Breakpoints } from './Breakpoints';
export type { IBreakpoints } from './Breakpoints';
export type { AbsoluteUnits, RelativeUnits, Units } from './units';
export { default as Sizes } from './Sizes';
export type { ISizes } from './Sizes';
