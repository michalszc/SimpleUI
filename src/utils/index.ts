export * from "./types";
export type { StyleProps } from "./styles";
export * from './theme';
export * from './convertColors';