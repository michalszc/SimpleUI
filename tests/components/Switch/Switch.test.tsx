import React from "react";
import '@testing-library/jest-dom';
import { fireEvent, render } from "@testing-library/react";
import { Switch } from "../../../src";

describe("Switch", () => {
    const mockFn = jest.fn();
    beforeEach(() => {
        mockFn.mockClear();
    });
    test("should properly render Switch component - default", () => {
        const container = render(<Switch />);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Switch component - checked", () => {
        const container = render(<Switch isChecked={true}/>);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Switch component - shape circle, enabled", () => {
        const container = render(<Switch shape={'circle'} isDisabled={false}/>);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Switch component - shape square, enabled", () => {
        const container = render(<Switch shape={'square'} isDisabled={false}/>);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Switch component - shape roundedSquare, enabled", () => {
        const container = render(<Switch shape={'roundedSquare'} isDisabled={false}/>);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Switch component - shape circle, disabled", () => {
        const container = render(<Switch shape={'circle'} isDisabled={true}/>);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Switch component - shape square, disabled", () => {
        const container = render(<Switch shape={'square'} isDisabled={true}/>);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Switch component - shape roundedSquare, disabled", () => {
        const container = render(<Switch shape={'roundedSquare'} isDisabled={true}/>);
        expect(container).toMatchSnapshot();
    });
    test("should call onChange function on click", () => {
        const { getByTestId } = render(<Switch onChange={mockFn} />);
        const label = getByTestId('test-switchcontainer');
        fireEvent.click(label);
        expect(mockFn).toBeCalled();
    });
    test("should not call onChange function on click - disabled", () => {
        const { getByTestId } = render(<Switch onChange={mockFn} isDisabled={true} />);
        const label = getByTestId('test-switchcontainer');
        fireEvent.click(label);
        expect(mockFn).not.toBeCalled();
    });
    test("should not call onChange function on click - readonly", () => {
        const { getByTestId } = render(<Switch onChange={mockFn} isReadonly={true} />);
        const label = getByTestId('test-switchcontainer');
        fireEvent.click(label);
        expect(mockFn).not.toBeCalled();
    });
});
