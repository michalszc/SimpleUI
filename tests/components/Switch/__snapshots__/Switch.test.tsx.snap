// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should properly render Switch component - checked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-hBxehG ggruCt simpleui-switch-container"
        data-testid="test-switchcontainer"
        id="4fzzzxjylrx"
        mode="light"
      >
        <input
          checked=""
          id="4fzzzxjylrx"
          name=""
          type="checkbox"
        />
        <span
          class="sc-fnGiBr ibGDdj simpleui-switch-slider"
          id="4fzzzxjylrx"
          mode="light"
          shape="circle"
        >
          <span
            class="sc-fEXmlR fDNaFm simpleui-switch-dot"
            id="4fzzzxjylrx"
            mode="light"
            shape="circle"
          />
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-hBxehG ggruCt simpleui-switch-container"
      data-testid="test-switchcontainer"
      id="4fzzzxjylrx"
      mode="light"
    >
      <input
        checked=""
        id="4fzzzxjylrx"
        name=""
        type="checkbox"
      />
      <span
        class="sc-fnGiBr ibGDdj simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="circle"
      >
        <span
          class="sc-fEXmlR fDNaFm simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="circle"
        />
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch should properly render Switch component - default 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-hBxehG ggruCt simpleui-switch-container"
        data-testid="test-switchcontainer"
        id="4fzzzxjylrx"
        mode="light"
      >
        <input
          id="4fzzzxjylrx"
          name=""
          type="checkbox"
        />
        <span
          class="sc-fnGiBr ibGDdj simpleui-switch-slider"
          id="4fzzzxjylrx"
          mode="light"
          shape="circle"
        >
          <span
            class="sc-fEXmlR fDNaFm simpleui-switch-dot"
            id="4fzzzxjylrx"
            mode="light"
            shape="circle"
          />
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-hBxehG ggruCt simpleui-switch-container"
      data-testid="test-switchcontainer"
      id="4fzzzxjylrx"
      mode="light"
    >
      <input
        id="4fzzzxjylrx"
        name=""
        type="checkbox"
      />
      <span
        class="sc-fnGiBr ibGDdj simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="circle"
      >
        <span
          class="sc-fEXmlR fDNaFm simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="circle"
        />
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch should properly render Switch component - shape circle, disabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-hBxehG ggruCt simpleui-switch-container"
        data-testid="test-switchcontainer"
        id="4fzzzxjylrx"
        mode="light"
      >
        <input
          disabled=""
          id="4fzzzxjylrx"
          name=""
          type="checkbox"
        />
        <span
          class="sc-fnGiBr pWgSJ simpleui-switch-slider"
          id="4fzzzxjylrx"
          mode="light"
          shape="circle"
        >
          <span
            class="sc-fEXmlR fDNaFm simpleui-switch-dot"
            id="4fzzzxjylrx"
            mode="light"
            shape="circle"
          />
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-hBxehG ggruCt simpleui-switch-container"
      data-testid="test-switchcontainer"
      id="4fzzzxjylrx"
      mode="light"
    >
      <input
        disabled=""
        id="4fzzzxjylrx"
        name=""
        type="checkbox"
      />
      <span
        class="sc-fnGiBr pWgSJ simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="circle"
      >
        <span
          class="sc-fEXmlR fDNaFm simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="circle"
        />
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch should properly render Switch component - shape circle, enabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-hBxehG ggruCt simpleui-switch-container"
        data-testid="test-switchcontainer"
        id="4fzzzxjylrx"
        mode="light"
      >
        <input
          id="4fzzzxjylrx"
          name=""
          type="checkbox"
        />
        <span
          class="sc-fnGiBr ibGDdj simpleui-switch-slider"
          id="4fzzzxjylrx"
          mode="light"
          shape="circle"
        >
          <span
            class="sc-fEXmlR fDNaFm simpleui-switch-dot"
            id="4fzzzxjylrx"
            mode="light"
            shape="circle"
          />
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-hBxehG ggruCt simpleui-switch-container"
      data-testid="test-switchcontainer"
      id="4fzzzxjylrx"
      mode="light"
    >
      <input
        id="4fzzzxjylrx"
        name=""
        type="checkbox"
      />
      <span
        class="sc-fnGiBr ibGDdj simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="circle"
      >
        <span
          class="sc-fEXmlR fDNaFm simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="circle"
        />
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch should properly render Switch component - shape roundedSquare, disabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-hBxehG ggruCt simpleui-switch-container"
        data-testid="test-switchcontainer"
        id="4fzzzxjylrx"
        mode="light"
      >
        <input
          disabled=""
          id="4fzzzxjylrx"
          name=""
          type="checkbox"
        />
        <span
          class="sc-fnGiBr kVfPLL simpleui-switch-slider"
          id="4fzzzxjylrx"
          mode="light"
          shape="roundedSquare"
        >
          <span
            class="sc-fEXmlR awDfN simpleui-switch-dot"
            id="4fzzzxjylrx"
            mode="light"
            shape="roundedSquare"
          />
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-hBxehG ggruCt simpleui-switch-container"
      data-testid="test-switchcontainer"
      id="4fzzzxjylrx"
      mode="light"
    >
      <input
        disabled=""
        id="4fzzzxjylrx"
        name=""
        type="checkbox"
      />
      <span
        class="sc-fnGiBr kVfPLL simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="roundedSquare"
      >
        <span
          class="sc-fEXmlR awDfN simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="roundedSquare"
        />
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch should properly render Switch component - shape roundedSquare, enabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-hBxehG ggruCt simpleui-switch-container"
        data-testid="test-switchcontainer"
        id="4fzzzxjylrx"
        mode="light"
      >
        <input
          id="4fzzzxjylrx"
          name=""
          type="checkbox"
        />
        <span
          class="sc-fnGiBr gqTWDB simpleui-switch-slider"
          id="4fzzzxjylrx"
          mode="light"
          shape="roundedSquare"
        >
          <span
            class="sc-fEXmlR awDfN simpleui-switch-dot"
            id="4fzzzxjylrx"
            mode="light"
            shape="roundedSquare"
          />
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-hBxehG ggruCt simpleui-switch-container"
      data-testid="test-switchcontainer"
      id="4fzzzxjylrx"
      mode="light"
    >
      <input
        id="4fzzzxjylrx"
        name=""
        type="checkbox"
      />
      <span
        class="sc-fnGiBr gqTWDB simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="roundedSquare"
      >
        <span
          class="sc-fEXmlR awDfN simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="roundedSquare"
        />
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch should properly render Switch component - shape square, disabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-hBxehG ggruCt simpleui-switch-container"
        data-testid="test-switchcontainer"
        id="4fzzzxjylrx"
        mode="light"
      >
        <input
          disabled=""
          id="4fzzzxjylrx"
          name=""
          type="checkbox"
        />
        <span
          class="sc-fnGiBr hYuXni simpleui-switch-slider"
          id="4fzzzxjylrx"
          mode="light"
          shape="square"
        >
          <span
            class="sc-fEXmlR gAcZqy simpleui-switch-dot"
            id="4fzzzxjylrx"
            mode="light"
            shape="square"
          />
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-hBxehG ggruCt simpleui-switch-container"
      data-testid="test-switchcontainer"
      id="4fzzzxjylrx"
      mode="light"
    >
      <input
        disabled=""
        id="4fzzzxjylrx"
        name=""
        type="checkbox"
      />
      <span
        class="sc-fnGiBr hYuXni simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="square"
      >
        <span
          class="sc-fEXmlR gAcZqy simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="square"
        />
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Switch should properly render Switch component - shape square, enabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        class="sc-hBxehG ggruCt simpleui-switch-container"
        data-testid="test-switchcontainer"
        id="4fzzzxjylrx"
        mode="light"
      >
        <input
          id="4fzzzxjylrx"
          name=""
          type="checkbox"
        />
        <span
          class="sc-fnGiBr blLJPw simpleui-switch-slider"
          id="4fzzzxjylrx"
          mode="light"
          shape="square"
        >
          <span
            class="sc-fEXmlR gAcZqy simpleui-switch-dot"
            id="4fzzzxjylrx"
            mode="light"
            shape="square"
          />
        </span>
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="sc-hBxehG ggruCt simpleui-switch-container"
      data-testid="test-switchcontainer"
      id="4fzzzxjylrx"
      mode="light"
    >
      <input
        id="4fzzzxjylrx"
        name=""
        type="checkbox"
      />
      <span
        class="sc-fnGiBr blLJPw simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="square"
      >
        <span
          class="sc-fEXmlR gAcZqy simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="square"
        />
      </span>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
